src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,259 @@
+    this.input = input.current;
+  };
+
+  blockBlur() {
+    this.blurBlocked = true;
+    this.bridge.current.blockBlur = true; // this is to support the legacy behaviour in the bridge
+  }
+
+  unblockBlur() {
+    this.blurBlocked = false;
+    this.bridge.current.blockBlur = false; // this is to support the legacy behaviour in the bridge
+  }
+
+  handleBlur = (ev) => {
+    if (this.blurBlocked) return;
+    if (this.props.onBlur) this.props.onBlur(ev);
+  }
+
+  handleFocus = (ev) => {
+    if (this.blockFocus) {
+      this.blockFocus = false;
+    } else {
+      this.openDatePicker();
+    }
+
+    if (this.props.onFocus) this.props.onFocus(ev);
+  };
+
+  handleKeyDown = (ev) => {
+    if (Events.isTabKey(ev)) {
+      this.closeDatePicker();
+    }
+  };
+
+  openDatePicker = () => {
+    const isValidDate = DateHelper.isValidDate(this.props.value);
+
+    this.blockBlur();
+    this._document.addEventListener('click', this.closeDatePicker);
+
+    if (isValidDate) {
+      this.updateDatePickerValue(this.props.value);
+    }
+
+    this.setState({ isDatePickerOpen: true });
+  };
+
+  closeDatePicker = () => {
+    this._document.removeEventListener('click', this.closeDatePicker);
+    this.setState((prevState) => {
+      return {
+        visibleValue: this.formatVisibleValue(prevState.selectedDate),
+        isDatePickerOpen: false
+      };
+    });
+  };
+
+  handleDateSelect = (selectedDate, modifiers) => {
+    const stringDateIso = DateHelper.formatDateString(selectedDate, isoDateFormat);
+
+    if (modifiers.disabled) return;
+
+    this.blockBlur();
+    this.closeDatePicker();
+    this.updateVisibleValue(selectedDate);
+    this.emitOnChangeCallback(stringDateIso);
+  };
+
+  updateVisibleValue = (date) => {
+    const formattedDateString = this.formatVisibleValue(date);
+    this.setState({ visibleValue: formattedDateString });
+  };
+
+  formatVisibleValue(value) {
+    // Don't sanitize so it accepts the hidden format (with dash separators)
+    return DateHelper.formatValue(value || today, getVisibleFormat(), {
+      formats: isoDateFormat,
+      sanitize: false
+    });
+  }
+
+  handleVisibleInputChange = (ev) => {
+    const dateWithSlashes = DateHelper.sanitizeDateInput(ev.target.value);
+    const isValidDate = DateHelper.isValidDate(dateWithSlashes);
+    const isoDateString = DateHelper.formatDateString(dateWithSlashes, isoDateFormat);
+
+    // Updates the hidden value after first formatting to default hidden format
+    if (isValidDate) {
+      this.updateDatePickerValue(dateWithSlashes);
+      this.emitOnChangeCallback(isoDateString);
+    }
+
+    this.setState({ visibleValue: ev.target.value });
+  };
+
+  updateDatePickerValue = (newValue) => {
+    const isoDate = convertToIsoDate(newValue);
+
+    this.setState({ selectedDate: isoDate });
+  };
+
+  emitOnChangeCallback = (stringDate) => {
+    this.props.onChange({ target: { value: stringDate } });
+  };
+
+  renderDatePicker = (passedProps) => {
```

Comment:
Perhaps give this param a more descriptive name, something like `minMax`.

Reply:
we could pass any prop later, for now it's only `minDate` and `maxDate`

Reply:
Which other props will be passed?

Reply:
we could pass other DatePicker config props needed to render a proper DatePicker, depending on the future requirements.

Reply:
Presumably that future requirement would require us to make changes to this file, though? For quick understanding of how the Component works currently, it might be better to give this a more descriptive name.

Reply:
what do you think about `dateLimitProps`?